### PR TITLE
Refactor Authentication Redirect Data

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request updates our authentication redirect helper to automatically include the pertinent contextual data for Northstar (with the option to override/embellish) while removing our reliance on Redux state.

We've also stopped using the Cover Image and description fields over in Northstar (https://github.com/DoSomething/northstar/pull/1038), so those are removed in https://github.com/DoSomething/phoenix-next/commit/5c741443dd64b43f1208e652392223138eb660d4

https://github.com/DoSomething/phoenix-next/commit/7ad37fad66a7fabae14f0614652bbb4de9569f8a adds a test case for the recent bug scenario in #2529 

### How should this be reviewed?
👀  commit-by-commit

### Any background context you want to provide?
As I work on refactoring our Campaign signup form, and look to include our `useGate` hook for authentication, I notice that [we don't include](https://github.com/DoSomething/phoenix-next/blob/085247915fad75b343ce31320517e858e5809234/resources/assets/helpers/auth.js#L130) contextual data currently. It feels like a win to just automatically include these values, and get rid of Redux reliance while we're at it!

### Relevant tickets

References [Pivotal #175291206](https://www.pivotaltracker.com/story/show/175291206).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/selectors/index.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+  return get(window.STATE, 'page.fields');
+};
 
 export function getDataForNorthstar(state) {
   return withoutValueless({
-    ...getCampaignDataForNorthstar(state),
-    ...getStoryPageDataForNorthstar(state),
-
+    // Used as 'destination' parameter for customized destination title on Northstar login page.
+    title: get(getCampaign(), 'title') || get(getStoryPage(), 'title'),
```
Comment (mendelB) [inline resources/assets/selectors/index.js]: We use this as the `destination` parameter [here](https://github.com/DoSomething/phoenix-next/blob/cce8107e841fffa0ace2caf16350a25d711ac915/app/Http/Controllers/AuthController.php#L69-L80)

File: resources/assets/components/pages/ReferralPage/Beta/BetaPageCampaignLink.js [hunk 2]
```diff
@@ -13,7 +13,6 @@ const REFERRAL_PAGE_CAMPAIGN = gql`
   query ReferralPageCampaignQuery($campaignId: String!, $preview: Boolean!) {
     campaignWebsiteByCampaignId(campaignId: $campaignId, preview: $preview) {
       id
-
```
Comment (DFurnes) [inline resources/assets/components/pages/ReferralPage/Beta/BetaPageCampaignLink.js:16]: This 3452ea101b078442e77eaafa61fe31e326184964 and 21438e3ed668df2071a3192612a3c182fd4576c0 commit pair genuinely cracked me up. 😂

File: resources/assets/helpers/auth.js [hunk 3]
```diff
@@ -52,6 +54,29 @@ export function bindTokenRefreshEvent() {
...
+
+  const storyPageTitle =
+    get(page, 'type') !== 'storyPage' && get(page, 'fields.title');
+
+  return withoutValueless({
+    // Used as 'destination' parameter for customized destination title on Northstar login page.
+    title: get(campaign, 'title') || storyPageTitle,
+    // For 'source_details':
+    contentful_id: campaign.id || page.id,
+    mode: query('mode') || null,
```
Comment (DFurnes) [inline resources/assets/helpers/auth.js:74]: Is this `mode` for whether to default to login or register flow? If so, should we add a note for that?

**Review by DFurnes — APPROVED** [review]
Amazing! 🍃 🔏

File: resources/assets/components/pages/ReferralPage/Beta/BetaPageCampaignLink.js (hunk 2 above)
Comment (mendelB) [reply resources/assets/components/pages/ReferralPage/Beta/BetaPageCampaignLink.js:16]: LOL https://pbfcomics.com/comics/negative-space/

File: resources/assets/helpers/auth.js (hunk 3 above)
Comment (mendelB) [reply resources/assets/helpers/auth.js:75]: Yes! Added in https://github.com/DoSomething/phoenix-next/pull/2532/commits/b2b71fd652305b449c1c8dcac91786789448993a

**Pull request merged**